src/App.vue

```diff
@@ -502,6 +504,8 @@ export default {
 
 		if (this.$route.hash === '#direct-call') {
 			emit('talk:media-settings:show')
+		} else if (this.$route.hash === '#settings') {
+			emit('show-conversation-settings', { token: this.token })
```

Comment:
This feature is a conversation-only feature, but handled here on a whole app level:
- It handles hash and emits conversation-related event on all the pages without conversation (token)
- Though it works right now, there is no guarantee conversation components are mounted and ready at this moment. It will be broken if we add anything async to the conversation page init.
- It uses `settings` fragment, leaving no tinny name for Talk settings if we add it later

Reply:
What about moving it to the conversation view?

Reply:
And about removing the hash from URL when the settings dialog is open

Reply:
- ConversationSettingsDialog is detached and independent from conversations. So it could be called on empty view as well, contrary to `#direct-call`
- It requires only token+conversation object to work. Maybe some additional request is needed here to ensure we fetched a conversation prior to opening the dialog
- Removing of hash should be fine, as long as it doesn't triggering any vue-router side effects

Reply:
> * as long as it doesn't triggering any vue-router side effects

If it does, it is a bug that should be fixed

Reply:
> It requires only token+conversation object to work

We don't have token+conversation outside the conversation view.
And a fragment cannot specify it. If we want to add ability to open setting for any conversation, it must be a query param. (Or if we want it to be combined with other fragments, like message id).

Reply:
/apps/spreed/#settings

![image](https://github.com/user-attachments/assets/ca956aa8-eda8-4649-8bf8-2e7ccd16e7f5)